scripts/hail_batch/variant_selection/hgdp_1kg_tob_wgs_variant_selection.py

```diff
@@ -0,0 +1,70 @@
+"""Pipeline for choosing new variants for HGDP/1kG + TOB-WGS data"""
+
+import click
+import hail as hl
+from hail.experimental import lgt_to_gt
+
+GNOMAD_HGDP_1KG_MT = (
+    'gs://gcp-public-data--gnomad/release/3.1/mt/genomes/'
+    'gnomad.genomes.v3.1.hgdp_1kg_subset_dense.mt'
+)
+
+TOB_WGS = 'gs://cpg-tob-wgs-main/joint_vcf/v1/raw/genomes.mt'
+
+
+@click.command()
+@click.option('--output', help='GCS output path', required=True)
+def query(output):  # pylint: disable=too-many-locals
+    """Query script entry point."""
+
+    hl.init(default_reference='GRCh38')
+
+    hgdp_1kg = hl.read_matrix_table(GNOMAD_HGDP_1KG_MT)
+    tob_wgs = hl.read_matrix_table(TOB_WGS).key_rows_by('locus', 'alleles')
+
+    # filter to loci that are contained in both matrix tables
+    hgdp_1kg = hgdp_1kg.filter_rows(
+        hl.is_defined(tob_wgs.index_rows(hgdp_1kg['locus'], hgdp_1kg['alleles']))
+    )
+    tob_wgs = tob_wgs.semi_join_rows(hgdp_1kg.rows())
+
+    # Entries and columns must be identical
+    tob_wgs_select = tob_wgs.select_entries(GT=lgt_to_gt(tob_wgs.LGT, tob_wgs.LA))
+    hgdp_1kg_select = hgdp_1kg.select_entries(hgdp_1kg.GT)
+    hgdp_1kg_select = hgdp_1kg_select.select_cols()
+    # Join datasets
+    hgdp1kg_tobwgs_joined = hgdp_1kg_select.union_cols(tob_wgs_select)
+    # Add in metadata information
+    hgdp_1kg_metadata = hgdp_1kg.cols()
+    hgdp1kg_tobwgs_joined = hgdp1kg_tobwgs_joined.annotate_cols(
+        hgdp_1kg_metadata=hgdp_1kg_metadata[hgdp1kg_tobwgs_joined.s]
+    )
+    mt_path = f'{output}/hgdp1kg_tobwgs_joined_all_samples.mt'
+    if not hl.hadoop_exists(mt_path):
+        hgdp1kg_tobwgs_joined.write(mt_path)
```

Comment:
Do we really need to duplicate a significant fraction of the 1KG dataset here?

Reply:
no, you're right we don't necessarily need this. Perhaps just take it out?